src/dataflow-types/src/types.rs

```diff
@@ -626,6 +627,42 @@ pub struct FileSourceConnector {
     pub tail: bool,
 }
 
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct S3SourceConnector {
+    pub bucket: String,
+    /// Used to filter results
+    #[serde(with = "s3_serde_glob")]
+    pub objects_pattern: glob::Pattern,
+
+    /// Information required to make any rusoto AWS connection
+    pub aws_info: AwsConnectInfo,
+}
+
+mod s3_serde_glob {
+    pub(crate) fn serialize<S: serde::Serializer>(
+        pattern: &glob::Pattern,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        serializer.serialize_str(pattern.as_str())
+    }
+
+    pub(crate) fn deserialize<'de, D>(d: D) -> Result<glob::Pattern, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        use serde::Deserialize;
+        glob::Pattern::new(&String::deserialize(d)?).map_err(|e| serde::de::Error::custom(e))
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct AwsConnectInfo {
```

Comment:
This is slightly different to a connector, though, since it's a subset of the information required by connectors, it's not enough info by itself to create a connector, and I felt like that difference was worth preserving. I agree that `AwsConnectInfo` is a bad name, but I couldn't think of anything better.

Reply:
I have to admit I'm unconvinced! The term "connector" was never meant to mean "complete description"; it was picked specifically to avoid the noise word "info".
And indeed an `ExternalSourceConnector` is not actually a complete of what you need to create an external source—there are a bunch of fields in `SourceConnector::External`. 

Anyway, minor, let's definitely not hold up this PR on this point!